Build-Dependencies.ps1

```diff
@@ -0,0 +1,176 @@
+[CmdletBinding()]
+param(
+    [ValidateSet('x86', 'x64')]
+    [string] $Target,
+    [ValidateSet('Debug', 'RelWithDebInfo', 'Release', 'MinSizeRel')]
+    [string] $Configuration = 'Release',
+    [string[]] $Dependencies,
+    [switch] $SkipAll,
+    [switch] $SkipDeps,
+    [switch] $SkipUnpack,
+    [switch] $SkipBuild,
+    [switch] $Shared,
+    [switch] $Clean,
+    [switch] $Quiet
+)
+
+$ErrorActionPreference = "Stop"
+
+if ( $DebugPreference -eq "Continue" ) {
+    $VerbosePreference = "Continue"
+    $InformationPreference = "Continue"
+}
+
+function Run-Stages {
+    $Stages = @('Setup')
+
+    if ( ( $script:SkipAll.isPresent ) -or ( $script:SkipBuild.isPresent ) ) {
```

Comment:
Yeah I removed passing variables into imported scripts to reduce code complexity and explicitly use the `script` scope to access the value of switches passed to the main build scripts.

As such `$script:SkipAll` is correct, but per the docs `isPresent` could be removed.

Reply:
Looks like removing `isPresent` breaks the script in some places, but not others. Looks like its return value is more compatible that whatever the variable itself coalesces to. I reverted the changes accordingly.

Reply:
Per the docs:
> Switch parameters are of type `SwitchParameter`, which implicitly converts to Boolean.

What's interesting is that `.IsPresent` is meant only for SwitchParameters, and a $script scope variable is not necessarily a Switch Parameter, so this should be an unreliable call as-is.

I think the more reliable method in cases like this would be to use the $script scope variable to set the default value of the Switch Parameter of this file like we do in the current build scripts:
```
[switch] $SkipAll = $(if (Test-Path Variable:SkipAll) { $SkipAll }),
```

Reply:
Correct, the switch is defined in this file, so the `script` prefix is unnecessary here. In dot-included files the prefix has to be used as it needs to "inherit" the switch value from the calling script scope (and it seems that only the return value of `IsPresent` can be used as an array index).